### PR TITLE
(FACT-855) Rename cfacter to facter

### PR DESCRIPTION
**Pull request by MikaelSmith**
As part of the move to facter/master, the name cfacter is being retired.
Replace all references to cfacter with facter.

The only exception is contrib/cfacter.rb for Homebrew, which needs to
wait for a release to be renamed.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
This change is required for running acceptance tests.

**Comment by whopper** [comment]
:+1: Facter builds with these changes and runs correctly! Woo!

File: ext/redhat/facter.spec.erb [hunk 1]
```diff
@@ -94,11 +94,11 @@ rm -rf %{buildroot}
 
 %files
 %defattr(-,root,root,-)
-%{_bindir}/cfacter
+%{_bindir}/facter
 %{_libdir}/libfacter.so*
 %{_includedir}/facter
 %if 0%{?rhel} >= 7 || 0%{?fedora}
-%{vendor_ruby}/cfacter.rb
+%{vendor_ruby}/facter.rb
```
Comment (peterhuene) [inline ext/redhat/facter.spec.erb]: I'm debating whether we need a cfacter.rb to maintain the `--cfacter` option in puppet or if we treat the `--cfacter` option as only supported with "cfacter" (which Facter 3 is not).

File: lib/facter.rb.in [hunk 2]
```diff
@@ -11,7 +11,7 @@ module CFacter
   # @api public
   def self.initialize
     begin
-      require "#{ENV['CFACTER_DIR'] || '${CMAKE_INSTALL_PREFIX}'}/${LIBFACTER_INSTALL_DESTINATION}/libfacter.so"
+      require "#{ENV['FACTER_DIR'] || '${CMAKE_INSTALL_PREFIX}'}/${LIBFACTER_INSTALL_DESTINATION}/libfacter.so"
```
Comment (peterhuene) [inline lib/facter.rb.in]: If we want to behave as `facter.rb`, we shouldn't have an `initialize` method, but instead initialize libfacter automatically.  Thus, users that are doing `require 'facter'` get the expected API without having to do `Facter.initialize`.

**Comment by peterhuene** [comment]
Other than those comments, :+1:

File: ext/redhat/facter.spec.erb (hunk 1 above)
Comment (MikaelSmith) [inline ext/redhat/facter.spec.erb]: I haven't really delved into that yet. Puppet will need to support either Facter 2 or Facter 3 though.

File: lib/facter.rb.in (hunk 2 above)
Comment (peterhuene) [inline lib/facter.rb.in]: Also, if we change this, we should update the README.

File: ext/redhat/facter.spec.erb (hunk 1 above)
Comment (MikaelSmith) [inline ext/redhat/facter.spec.erb]: Oh, got it. Do we treat Facter 3 as a drop-in replacement for Facter 2 against Puppet 4.0? Not sure either, I don't have a strong reason for doing it.

Comment (peterhuene) [inline ext/redhat/facter.spec.erb]: I'd probably say yes to that, in which case we really just treat the `--cfacter` option as something that worked with cfacter 0.2.0 to 0.4.0.

**Comment by MikaelSmith** [comment]
Updated.

File: ext/redhat/facter.spec.erb (hunk 1 above)
Comment (kylog) [inline ext/redhat/facter.spec.erb]: So we don't need a `cfacter.rb` anymore, and the `--cfacter` is only for those *cfacter* versions.

:+1:

**Comment by peterhuene** [comment]
:+1:

**Pull request merged**